frontend/src/features/collections/edit-dialog/general-section.ts

```diff
@@ -0,0 +1,320 @@
+import { msg } from "@lit/localize";
+import { TaskStatus } from "@lit/task";
+import { type SlInput } from "@shoelace-style/shoelace";
+import clsx from "clsx";
+import { html, nothing } from "lit";
+import { isEqual } from "lodash";
+import queryString from "query-string";
+
+import {
+  validateCaptionMax,
+  validateNameMax,
+  type CollectionEdit,
+} from "../collection-edit-dialog";
+import { type BtrixValidateDetails } from "../collection-snapshot-preview";
+import {
+  CollectionThumbnail,
+  DEFAULT_THUMBNAIL_VARIANT,
+  Thumbnail,
+} from "../collection-thumbnail";
+import { type SelectSnapshotDetail } from "../select-collection-page";
+
+import { snapshotToSource, sourceToSnapshot } from "./helpers/snapshots";
+
+import type { PublicCollection } from "@/types/collection";
+import { tw } from "@/utils/tailwind";
+
+export default function renderGeneral(this: CollectionEdit) {
+  if (!this.collection) return;
+  return html`<sl-input
+      class="with-max-help-text"
+      name="name"
+      label=${msg("Name")}
+      value=${this.collection.name}
+      placeholder=${msg("My Collection")}
+      autocomplete="off"
+      required
+      help-text=${validateNameMax.helpText}
+      @sl-input=${(e: CustomEvent) => {
+        this.validate(validateNameMax)(e);
+        this.name = (e.target as SlInput).value;
+      }}
+    >
+    </sl-input>
+    <sl-input
+      class="with-max-help-text"
+      name="caption"
+      value=${this.collection.caption ?? ""}
+      placeholder=${msg("Summarize the collection's content")}
+      autocomplete="off"
+      help-text=${validateCaptionMax.helpText}
+      @sl-input=${this.validate(validateCaptionMax)}
+    >
+      <span slot="label">
+        ${msg("Summary")}
+        <sl-tooltip>
+          <span slot="content">
+            ${msg(
+              "Write a short description that summarizes this collection. If the collection is public, this description will be visible next to the collection name.",
+            )}
+          </span>
+          <sl-icon name="info-circle" style="vertical-align: -.175em"></sl-icon>
+        </sl-tooltip>
+      </span>
+    </sl-input>
+    <div class="mb-7">${renderThumbnails.bind(this)()}</div>
+    <section>
+      <btrix-select-collection-page
+        mode="thumbnail"
+        .collection=${this.collection}
+        .collectionId=${this.collection.id}
+        .selectedSnapshot=${sourceToSnapshot(this.selectedSnapshot) ??
+        undefined}
+        @btrix-select=${async (e: CustomEvent<SelectSnapshotDetail>) => {
+          if (!e.detail.item) return;
+          const newSnapshot = snapshotToSource(e.detail.item);
+          if (!isEqual(newSnapshot, this.selectedSnapshot)) {
+            this.thumbnailIsValid = null;
+            this.selectedSnapshot = newSnapshot;
+          }
+
+          void this.checkChanged();
```

Comment:
If a URL that's not in the collection is typed in, it doesn't trigger the changes check. That behavior seems correct, but then I'm able to close out of the dialog, and when it's reopened the old value persists. We'll probably want to reset the form fields on `sl-after-hide`. Clicking "Cancel" also doesn't reset the Page URL field.